packages/dds/tree/src/shared-tree/treeApiAlpha.ts

```diff
@@ -49,9 +50,93 @@ import {
 	TreeCompressionStrategy,
 	type FieldBatch,
 	type FieldBatchEncodingContext,
+	type LocalNodeIdentifier,
 } from "../feature-libraries/index.js";
 import { independentInitializedView, type ViewContent } from "./independentView.js";
 import { SchematizingSimpleTreeView, ViewSlot } from "./schematizingTreeView.js";
+import { treeApi } from "./treeApi.js";
+
+const identifier: TreeIdentifierUtils = (node: TreeNode): string | undefined => {
+	const nodeIdentifier = getIdentifierFromNode(node, false);
+	if (typeof nodeIdentifier === "number") {
+		throw new TypeError("identifier should be uncompressed.");
+	}
+	return nodeIdentifier;
+};
+
+identifier.shorten = (branch: TreeBranch, nodeIdentifier: string): number | undefined => {
+	const nodeKeyManager = (branch as SchematizingSimpleTreeView<ImplicitFieldSchema>)
+		.nodeKeyManager;
+	const localNodeKey = nodeKeyManager.tryLocalizeNodeIdentifier(nodeIdentifier);
+	return localNodeKey !== undefined ? extractFromOpaque(localNodeKey) : undefined;
+};
+
+identifier.expand = (branch: TreeBranch, nodeIdentifier: number): string => {
+	const nodeKeyManager = (branch as SchematizingSimpleTreeView<ImplicitFieldSchema>)
+		.nodeKeyManager;
+	return nodeKeyManager.stabilizeNodeIdentifier(
+		nodeIdentifier as unknown as LocalNodeIdentifier,
+	);
+};
+
+identifier.getShort = (node: TreeNode): number | string | undefined => {
+	return treeApi.shortId(node);
+};
+
+Object.freeze(identifier);
```

Comment:
Interesting! I wonder if there are other API objects that we ought to be doing this to.